applications/asset_tracker_v2/src/modules/data_module.c

```diff
@@ -263,6 +269,76 @@ static bool app_event_handler(const struct app_event_header *aeh)
 	return false;
 }
 
+static bool grant_send(enum coneval_supported_data_type type)
+{
+#if defined(CONFIG_DATA_GRANT_SEND_ON_CONNECTION_QUALITY)
+	int err;
+	struct lte_lc_conn_eval_params coneval = { 0 };
+
+	if (send_denied_count[type] >= CONFIG_DATA_SEND_ATTEMPTS_COUNT_MAX) {
+		/* Grant send if a message has been attempted too many times. */
+		LOG_WRN("Too many attempts, granting send");
+		goto grant;
+	}
+
+	err = lte_lc_conn_eval_params_get(&coneval);
```

Comment:
The current implementation calls `grant_send()` three times in a row and each call triggers connection evaluation. Each connection evaluation consumes energy, so the implementation should be modified so, that connection evaluation is done only once and the same results are used for all data types.

Reply:
Agreed!